### PR TITLE
now that go test runs iteration loops, use that instead of custom executor

### PR DESCRIPTION
**Pull request by mikedanese**
cc @ixdy

### CODE REVIEW

File: hack/test-go.sh [hunk 1]
```diff
@@ -180,49 +175,9 @@ produceJUnitXMLReport() {
...
-      fi
-      count=$((count + 1))
-    done 2>&1
-    kube::log::status "Worker ${worker}: ${pass} / ${count} passed"
-  done
-  return 0
-}
-
 runTests() {
   # TODO: this should probably be refactored to avoid code duplication with the
```
Comment (ixdy) [inline hack/test-go.sh]: remove this comment too?

Comment (mikedanese) [inline hack/test-go.sh]: Fixed

**Comment by ixdy** [comment]
LGTM

**Comment by ixdy** [comment]
@k8s-bot test this please, issue #IGNORE 

(test results are pretty old at this point)

**Comment by ixdy** [comment]
@k8s-bot node e2e test this please, issue #IGNORE

(reconfigured Jenkins job, want to make sure I didn't break everything)

**Comment by ixdy** [comment]
@k8s-bot node e2e test this please, issue #IGNORE

yeah oops

**Pull request merged**